BuildingBlocks/test/BuildingBlocks.Application.Tests/Mediatr/QuotaEnforcerBehaviorTests.cs

```diff
@@ -1,100 +1,109 @@
 ﻿using Enmeshed.BuildingBlocks.Application.Abstractions.Exceptions;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.UserContext;
 using Enmeshed.BuildingBlocks.Application.Attributes;
 using Enmeshed.BuildingBlocks.Application.MediatR;
 using Enmeshed.BuildingBlocks.Domain;
 using FluentAssertions;
 using MediatR;
 using Xunit;
 
+using DomainMetricStatus = Enmeshed.BuildingBlocks.Domain.MetricStatus;
+
 namespace Enmeshed.BuildingBlocks.Application.Tests.Mediatr;
 public class QuotaEnforcerBehaviorTests
 {
-    public QuotaEnforcerBehaviorTests() { }
+    private readonly IUserContext _userContextStub;
+
+    public QuotaEnforcerBehaviorTests() {
+        _userContextStub = new UserContextStub();
+    }
 
-    [ApplyQuotasForMetrics("MetricKey")]
-    private class TestCommand : IRequest<IResponse> { }
+    [ApplyQuotasForMetrics("*")]
```

Comment:
One purpose of tests is documentation. When I am new to a project, the first thing I do is looking at the tests. 

In case of this test, if I would read this line without knowing anything about the implementation, I would probably assume that "*" means that quotas for ALL metrics should be applied.

That's probably even what you intended to express, because the tests don't care about what you pass as a Metric. They always return the same Metric.

But as I said, someone without further knowledge would make the wrong assumption.

In order to avoid that, I'd suggest a comment that says something like "The Metric key doesn't matter for these tests". Now you think "But he said that comments are bad". Yes, but only comments that describe "What is done". Comments that describe "Why something is done" are totally fine - as long as it is impossible to describe it with code. And now that I'm thinking aobut it: in this case you may even pass a Metric key called "DoesntMatterForThisTest" to avoid the comment.